src/main/java/org/junitpioneer/jupiter/json/JsonSource.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.json;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+/**
+ * {@code @JsonSource} is an {@link ArgumentsSource} that parses the json and passes arguments to the parametrized test.
+ *
+ *
+ * <p>This annotation is {@link Repeatable}, to make it usable with {@link org.junitpioneer.jupiter.CartesianProductTest}.
+ * If used with {@link org.junit.jupiter.params.ParameterizedTest}, it can only be used once (because {@code ParameterizedTest}
+ * can only take a single {@link ArgumentsSource}). Using it more than once will throw an {@link IllegalArgumentException}.
+ * If used with {@link org.junitpioneer.jupiter.CartesianProductTest}, it can be repeated to provide arguments to
+ * more than one parameter.
+ * </p>
+ *
+ * <p>
+ * For more details and examples, see
+ * <a href="https://junit-pioneer.org/docs/cartesian-product/" target="_top">the documentation on <code>Cartesian product tests</code></a>
+ * </p>
+ *
+ * @see ArgumentsSource
+ * @see org.junit.jupiter.params.ParameterizedTest
+ * @see org.junitpioneer.jupiter.CartesianProductTest
+ * @since TBD
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@ArgumentsSource(JsonSourceArgumentsProvider.class)
+@Repeatable(JsonSource.JsonSources.class)
+public @interface JsonSource {
```

Comment:
This is confusing. JUnit 5 has `@CsvSource` which can take CSV as a `String` and `CsvFileSource` which takes CSV file names. I think following that, this should be `JsonFileSource`.

Reply:
You are right about this. It would make more sense this to be named `@JsonFileSource`. I didn't really think about the naming and took what was there in the issue

Reply:
I renamed this into `@JsonFileSource`.